openjpa-jdbc/src/main/java/org/apache/openjpa/jdbc/sql/SelectImpl.java

```diff
@@ -2711,7 +2711,7 @@ private String getColumnAlias(Column col, PathJoins pj) {
                 return alias + "_" + col;
             }
             alias = SelectImpl.toAlias(_sel.getTableIndex(col.getTable(), pj, false));
-            return (alias == null) ? null : alias + "." + col;
+            return (alias == null) ? null : alias + "." + _sel._dict.getNamingUtil().toDBName(col.toString());
```

Comment:
the new test exercise this line.
I must be honest, with Derby it does not fail the test,  even without this line, but without this change the integration tests with HerdDB fail

Reply:
you can likely create an assert on the select string directly - even using internals to instantiate it if it helps.